presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -100,11 +101,21 @@ public void getAllQueryInfo(
             proxyResponse(servletRequest, asyncResponse, xForwardedProto, uriInfo);
             return;
         }
+        if (limitFilter == null) {
+            limitFilter = dispatchManager.getMaxQueriesToReturn();
+        }
         QueryState expectedState = stateFilter == null ? null : QueryState.valueOf(stateFilter.toUpperCase(Locale.ENGLISH));
         ImmutableList.Builder<BasicQueryInfo> builder = new ImmutableList.Builder<>();
+        // TODO: With us having 'limit' now, it would be nice to have 'basic query info' sorted  by some time
+        // (creation time?) in the query list that we get from dispatch manager.
+        // At the moment I don't see how we can add DateTime member to 'basic query info' and not break something.
         for (BasicQueryInfo queryInfo : dispatchManager.getQueries()) {
             if (stateFilter == null || queryInfo.getState() == expectedState) {
                 builder.add(queryInfo);
+                --limitFilter;
```

Comment:
I had an impression that in the code base using `--` as suffix is preferred when there is no semantic difference. Maybe let's change to `limitFilter--`?

Reply:
In cpp --a is prefered, at least in the past. Don't know about java. :) I can change.

Reply:
Ah I was referring to my impression about Presto code preference :D